src/Module.Server/Rewards/CrpgRewardServer.cs

```diff
@@ -195,17 +210,78 @@ public override void OnRemoveBehavior()
             return;
         }
 
+        CalculatePlayerUpkeepCompensations();
+        CompensateTeamhitPlayers(userUpdates);
+
         // TODO: add retry mechanism (the endpoint need to be idempotent though).
         try
         {
-            var res = (await _crpgClient.UpdateUsersAsync(new CrpgGameUsersUpdateRequest { Updates = userUpdates })).Data!;
+            var res = (await _crpgClient.UpdateUsersAsync(new CrpgGameUsersUpdateRequest { Updates = userUpdates.Values.ToList() })).Data!;

```

Comment:
Updates is a list of update , compensate teamhitplayers should return a list of update you can append to crpg updates. It should not be void.

Reply:
Done